### PR TITLE
Build namespace packages

### PR DESCRIPTION
**Pull request by skrah**
First attempt at building conda dynd.nd and dynd.ndt packages.

### CODE REVIEW

**Comment by skrah** [comment]
Travis CI is apparently okay with utilizing two cores.  The build could be sped up further by changing the cmake files, but that should be another PR.

File: .travis.yml [hunk 1]
```diff
@@ -43,10 +43,14 @@ matrix:
...
       - git clone --depth=1 https://github.com/libdynd/libdynd.git
-      - python setup.py install
+      - python setup.py install --target=ndt -j2
+      - rm -rf build dist dynd.ndt.egg-info
+      - echo dynd.ndt >> requirements.txt
+      - python setup.py install --target=nd -j2
     script:
       - pushd ..
-      - python -c "import dynd; dynd.test(verbosity=2, exit=True)"
+      - python -m dynd.ndt.test
```
Comment (insertinterestingnamehere) [inline .travis.yml]: Nice use of executable modules.

File: appveyor.yml [hunk 2]
```diff
@@ -24,7 +24,10 @@ build_script:
   - conda update conda
   - conda install conda-build
   - ps: if(-not $env:APPVEYOR_PULL_REQUEST_NUMBER) { conda install anaconda-client }
-  - conda build conda.recipe --channel dynd/channel/dev
+  - conda build conda\ndt.recipe --channel dynd/channel/dev
+  - rd /q /s build dist dynd.ndt.egg-info
+  - echo dynd.ndt >> requirements.txt
```
Comment (insertinterestingnamehere) [inline appveyor.yml]: Why is this necessary? Also, this line doesn't do what you think it does. `echo` takes precedence over the redirect. Luckily the redirect doesn't have to be put at the end of the line, so `>> requirements.txt echo dynd.ndt` should do what you want.

File: conda/nd.recipe/meta.yaml [hunk 3]
```diff
@@ -1,13 +1,13 @@
 package:
-  name: dynd-python
+  name: dynd.nd
   version: 0.7.2
 
 build:
   number: {{environ.get('TRAVIS_BUILD_NUMBER', 0)}}    # [unix]
   number: {{environ.get('APPVEYOR_BUILD_NUMBER', 0)}}  # [win]
   rpaths:
```
Comment (insertinterestingnamehere) [inline conda/nd.recipe/meta.yaml]: the rpaths here should actually no longer be necessary. If they still are for some reason, it's a bug and we should address it in some other way.

File: appveyor.yml (hunk 2 above)
Comment (insertinterestingnamehere) [inline appveyor.yml]: Okay, so, thinking about this a bit more, I'd really like to avoid automatic edits to committed files that get run as a part of the build process. That's a recipe for accidental garbage getting put into the repo. Is there any possible way around that?

File: setup.py [hunk 4]
```diff
@@ -210,7 +212,10 @@ def run(self):
     if sys.platform != 'win32':
         cmake_command.append(source)
         self.spawn(cmake_command)
-        self.spawn(['make'])
+        if PARALLEL:
```
Comment (insertinterestingnamehere) [inline setup.py]: Having a parallel build done through setup.py on unix is great.
At some point we should make this use the `-MP` option on Windows as well, but it's harder to do that without interfering with a user-specified `CMAKE_CXX_FLAGS`.

**Comment by insertinterestingnamehere** [comment]
Aside from the few comments here, this looks great.

File: setup.py (hunk 4 above)
Comment (skrah) [inline setup.py]: I tried using ```msbuild /m:2 libdynd.sln``` (after running ```vcvarsall.bat```), but had a couple of problems. Is that supposed to work?

File: appveyor.yml (hunk 2 above)
Comment (skrah) [inline appveyor.yml]: Hmm, the redirection works here (I looked at https://www.microsoft.com/resources/documentation/windows/xp/all/proddocs/en-us/redirection.mspx?mfr=true):

```
C:\Users\stefan\dynd-python>type requirements.txt
numpy

C:\Users\stefan\dynd-python>echo dynd.ndt >> requirements.txt

C:\Users\stefan\dynd-python>type requirements.txt
numpy
dynd.ndt
```

Yes, we can add ```dynd.ndt``` to ```install_requires``` in setup.py, depending on the target.

File: setup.py [hunk 5]
```diff
@@ -103,8 +105,13 @@ def get_library_names(self, build_type):
     else:
       return ndt + nd
 
-target = Target(DIST_TARGET)
+  def get_requirements(self):
+    reqs = open('dev-requirements.txt').read().strip().split('\n')
+    if self.target != 'ndt':
+      reqs.append('dynd.ndt')
```
Comment (skrah) [inline setup.py]: This solution is apparently not possible:

https://travis-ci.org/libdynd/dynd-python/jobs/134413087#L4138

I thought that conda would ignore dependencies in setup.py.

**Comment by skrah** [comment]
I've moved the ```dynd.ndt``` requirement to setup.py, which in turn requires a conda workaround (https://github.com/conda/conda/issues/508).

This isn't relevant for this issue, but perhaps not all dev-requirements should go into ```install_requires``` in setup.py. I just took over the existing code.

File: setup.py (hunk 4 above)
Comment (insertinterestingnamehere) [inline setup.py]: Hmm. As far as I know that should work too.

File: appveyor.yml (hunk 2 above)
Comment (insertinterestingnamehere) [inline appveyor.yml]: Sorry about the batch misinformation there. I forgot that even though this doesn't always work it still works fine in most cases. http://stackoverflow.com/questions/7225630/how-to-echo-2-no-quotes-to-a-file-from-a-batch-script was the issue I was thinking of, and it doesn't apply in this case.

Thanks for handling this in setup.py.

**Comment by insertinterestingnamehere** [comment]
This looks great to me now. I'll merge later today unless anyone objects.
